src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -57,6 +58,13 @@
     
     abstract public SamReader open(final File file);
 
+    public SamReader open(final Path path) {
+        final SamInputResource r = SamInputResource.of(path);
+        final Path indexMaybe = SamFiles.findIndex(path);
+        if (indexMaybe != null) r.index(indexMaybe);
+        return open(r);
+    }
```

Comment:
What if the `Path` is a cram, and the client wants to call `referenceSequence()` on the factory? Do we have a plan to make that work?